### PR TITLE
ci: Run axe-core tests on appveyor

### LINKED ISSUE #519: Run CI on additional platforms incl. Windows
To catch issues like #516, we should run our tests on additional platforms, starting with Windows. We could use Sauce Labs through CircleCI for this. It would allow us to catch installation and test problems with minimal effort. https://circleci.com/docs/1.0/browser-testing-with-sauce-labs/

### PR DESCRIPTION
**Pull request by scurker**
Tests should be running end-to-end on appveyor, with a few exceptions:

* PhantomJS is timing out on some integration tests. It's likely we are doing something to break PhantomJS on Windows, but unfortunately Phantom is not providing any useful logs beyond timing out.
* A number of integration tests with webdriver are failing in IE11. ([see branch with failing tests](https://ci.appveyor.com/project/scurker/axe-core/builds/23236242))
* Tests that are failing only in IE are skipped

Related #1384 
Closes #519 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: Stephen

### CODE REVIEW

File: lib/core/utils/preload-cssom.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ axe.utils.preloadCssom = function preloadCssom({
 		return q;
 	}
 
-	const dynamicDoc = document.implementation.createHTMLDocument();
+	const dynamicDoc = document.implementation.createHTMLDocument('New Document');
```
Comment (stephenmathieson) [inline lib/core/utils/preload-cssom.js]: I thought @jkodu had already fixed this? 🤔

File: test/checks/lists/only-dlitems.js [hunk 2]
```diff
@@ -126,15 +127,18 @@ describe('only-dlitems', function() {
 		assert.deepEqual(checkContext._relatedNodes, []);
 	});
 
-	it('should return false if <link> is used along side dt', function() {
-		var checkArgs = checkSetup(
-			'<dl id="target"><link rel="stylesheet" href="theme.css"><dt>A list</dt></dl>'
-		);
+	(isIE11 ? it.skip : it)(
```
Comment (stephenmathieson) [inline test/checks/lists/only-dlitems.js]: Please add a comment explaining why this test is skipped in IE11.

File: test/checks/lists/only-listitems.js [hunk 3]
```diff
@@ -139,15 +140,18 @@ describe('only-listitems', function() {
 		);
 	});
 
-	it('should return false if <link> is used along side li', function() {
-		var checkArgs = checkSetup(
-			'<ol id="target"><link rel="stylesheet" href="theme.css"><li>A list</li></ol>'
-		);
+	(isIE11 ? it.skip : it)(
```
Comment (stephenmathieson) [inline test/checks/lists/only-listitems.js]: Please add a comment.

File: test/checks/mobile/css-orientation-lock.js [hunk 4]
```diff
@@ -218,26 +219,29 @@ describe('css-orientation-lock tests', function() {
...
-						sheet: getSheet(
-							SHEET_DATA.MEDIA_STYLE_ORIENTATION_WITH_TRANSFORM_ROTATE_90
-						)
-					}
-				]
-			}
-		);
-		assert.isFalse(actual);
-	});
+	(isIE11 ? it.skip : it)(
```
Comment (stephenmathieson) [inline test/checks/mobile/css-orientation-lock.js]: Comment.

File: test/commons/text/form-control-value.js [hunk 5]
```diff
@@ -119,29 +120,32 @@ describe('text.formControlValue', function() {
...
-				.querySelectorAll(axe._tree[0], '#fixture input')
-				.forEach(function(target) {
-					assert.equal(
-						nativeTextboxValue(target),
-						'',
-						'Expected no value for ' + target.actualNode.outerHTML
-					);
-				});
-		});
+		(isIE11 ? it.skip : it)(
```
Comment (stephenmathieson) [inline test/commons/text/form-control-value.js]: Here too.

File: test/integration/full/css-orientation-lock/violations.js [hunk 6]
```diff
@@ -38,39 +39,42 @@ describe('css-orientation-lock violations test', function() {
...
-			{
-				runOnly: {
-					type: 'rule',
-					values: ['css-orientation-lock']
-				}
-			},
-			function(err, res) {
-				assert.isNull(err);
-				assert.isDefined(res);
+	(isIE11 ? it.skip : it)(
```
Comment (stephenmathieson) [inline test/integration/full/css-orientation-lock/violations.js]: Here too

File: test/integration/full/landmark-banner-is-top-level/landmark-banner-is-top-level-pass.js [hunk 7]
```diff
@@ -21,7 +22,7 @@ describe('landmark-banner-is-top-level test pass', function() {
 	});
 
 	describe('passes', function() {
-		it('should find 3', function() {
+		(isIE11 ? it.skip : it)('should find 3', function() {
```
Comment (stephenmathieson) [inline test/integration/full/landmark-banner-is-top-level/landmark-banner-is-top-level-pass.js]: Here too

File: test/integration/full/landmark-contentinfo-is-top-level/landmark-contentinfo-is-top-level-pass.js [hunk 8]
```diff
@@ -26,7 +27,7 @@ describe('landmark-contentinfo-is-top-level test pass', function() {
 	});
 
 	describe('passes', function() {
-		it('should find 2', function() {
+		(isIE11 ? it.skip : it)('should find 2', function() {
```
Comment (stephenmathieson) [inline test/integration/full/landmark-contentinfo-is-top-level/landmark-contentinfo-is-top-level-pass.js]: Here too

File: test/integration/full/preload-cssom/preload-cssom.js [hunk 9]
```diff
@@ -150,7 +151,7 @@ describe('preload cssom integration test', function() {
 		var shadowFixture;
 
 		before(function() {
-			if (isPhantom) {
+			if (isPhantom || isIE11) {
```
Comment (stephenmathieson) [inline test/integration/full/preload-cssom/preload-cssom.js]: Here too

Comment (stephenmathieson) [reply test/integration/full/preload-cssom/preload-cssom.js]: Also the others below...

File: test/rule-matches/autocomplete-matches.js [hunk 10]
```diff
@@ -146,7 +146,7 @@ describe('autocomplete-matches', function() {
 		elm.setAttribute('tabindex', -1);
 		elm.setAttribute('style', 'position:absolute; top:-9999em');
 
-		parent = document.createElement('div');
+		var parent = document.createElement('div');
```
Comment (stephenmathieson) [inline test/rule-matches/autocomplete-matches.js]: 😱 nice catch

File: lib/core/utils/preload-cssom.js (hunk 1 above)
Comment (scurker) [reply lib/core/utils/preload-cssom.js]: He probably did, I probably need to rebase on develop again.

File: test/rule-matches/autocomplete-matches.js (hunk 10 above)
Comment (scurker) [reply test/rule-matches/autocomplete-matches.js]: 💯

File: lib/core/utils/preload-cssom.js (hunk 1 above)
Comment (scurker) [reply lib/core/utils/preload-cssom.js]: Actually, looks like he has it for the test, but not for the util. 

https://github.com/dequelabs/axe-core/blob/08e4447d636e98fe2e7424ec856bc1b56ccaec94/test/checks/mobile/css-orientation-lock.js#L6-L8

Comment (jkodu) [reply lib/core/utils/preload-cssom.js]: I do have it for the util too - https://github.com/dequelabs/axe-core/pull/1339/files#diff-93d57bbe6c6d128e06d4fe309e06ff40R34

The PR is not merged yet.

**Pull request merged**